src/main/kotlin/snyk/common/CustomEndpoints.kt

```diff
@@ -0,0 +1,60 @@
+package snyk.common
+
+import java.net.URI
+
+fun toSnykCodeApiUrl(endpointUrl: String?): String {
+    val endpoint = resolveCustomEndpoint(endpointUrl)
+    val uri = URI(endpoint)
+
+    return when {
+        uri.isSaaS() -> endpoint.replace("https://", "https://deeproxy.").removeSuffix("api")
+        uri.isSingleTenant() -> endpoint.replace("registry-web", "deeproxy").removeSuffix("api")
+        else -> "https://deeproxy.snyk.io/"
+    }
+}
+
+fun toSnykCodeSettingsUrl(endpointUrl: String?): String {
```

Comment:
minor: we have that `endpointUrl` only for tests purposes, shell we make default value (from settings) for it then?

Reply:
I would prefer to inject the URL from outside, so make the function as "dumb" as possible and don't put IntelliJ dependencies (will be easier to mock, test etc).